# Tasca 5: Exploració de les dades

In [58]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)

## Exercici 1
__Descarrega el data set Airlines Delay: Airline on-time statistics and delay causes i carrega’l a un pandas Dataframe. Explora les dades que conté, i queda’t únicament amb les columnes que consideris rellevants.__

In [65]:
# Es carrega la base de dades en un Dataframe. 
# Es selecciona les tres primeres observacions per comprobar les dades que hi ha
data_airlines = pd.read_csv('DelayedFlights.csv')
data_airlines[:3]

Unnamed: 0  Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  \
0           0  2008      1           3          4   2003.0        1955   
1           1  2008      1           3          4    754.0         735   
2           2  2008      1           3          4    628.0         620   

   ArrTime  CRSArrTime UniqueCarrier  FlightNum TailNum  ActualElapsedTime  \
0   2211.0        2225            WN        335  N712SW              128.0   
1   1002.0        1000            WN       3231  N772SW              128.0   
2    804.0         750            WN        448  N428WN               96.0   

   CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  TaxiIn  \
0           150.0    116.0     -14.0       8.0    IAD  TPA       810     4.0   
1           145.0    113.0       2.0      19.0    IAD  TPA       810     5.0   
2            90.0     76.0      14.0       8.0    IND  BWI       515     3.0   

   TaxiOut  Cancelled CancellationCode  Diverted  CarrierDelay  WeatherDelay  \
0      8.0          0                N         0           NaN           NaN   
1     10.0          0                N         0           NaN           NaN   
2     17.0          0                N         0           NaN           NaN   

   NASDelay  SecurityDelay  LateAircraftDelay  
0       NaN            NaN                NaN  
1       NaN            NaN                NaN  
2       NaN            NaN                NaN

In [66]:
# Es fa un resum de les dades
data_airlines.describe()

Unnamed: 0       Year         Month    DayofMonth     DayOfWeek  \
count  1.936758e+06  1936758.0  1.936758e+06  1.936758e+06  1.936758e+06   
mean   3.341651e+06     2008.0  6.111106e+00  1.575347e+01  3.984827e+00   
std    2.066065e+06        0.0  3.482546e+00  8.776272e+00  1.995966e+00   
min    0.000000e+00     2008.0  1.000000e+00  1.000000e+00  1.000000e+00   
25%    1.517452e+06     2008.0  3.000000e+00  8.000000e+00  2.000000e+00   
50%    3.242558e+06     2008.0  6.000000e+00  1.600000e+01  4.000000e+00   
75%    4.972467e+06     2008.0  9.000000e+00  2.300000e+01  6.000000e+00   
max    7.009727e+06     2008.0  1.200000e+01  3.100000e+01  7.000000e+00   

            DepTime    CRSDepTime       ArrTime    CRSArrTime     FlightNum  \
count  1.936758e+06  1.936758e+06  1.929648e+06  1.936758e+06  1.936758e+06   
mean   1.518534e+03  1.467473e+03  1.610141e+03  1.634225e+03  2.184263e+03   
std    4.504853e+02  4.247668e+02  5.481781e+02  4.646347e+02  1.944702e+03   
min    1.000000e+00  0.000000e+00  1.000000e+00  0.000000e+00  1.000000e+00   
25%    1.203000e+03  1.135000e+03  1.316000e+03  1.325000e+03  6.100000e+02   
50%    1.545000e+03  1.510000e+03  1.715000e+03  1.705000e+03  1.543000e+03   
75%    1.900000e+03  1.815000e+03  2.030000e+03  2.014000e+03  3.422000e+03   
max    2.400000e+03  2.359000e+03  2.400000e+03  2.400000e+03  9.742000e+03   

       ActualElapsedTime  CRSElapsedTime       AirTime      ArrDelay  \
count       1.928371e+06    1.936560e+06  1.928371e+06  1.928371e+06   
mean        1.333059e+02    1.343027e+02  1.082771e+02  4.219988e+01   
std         7.206007e+01    7.134144e+01  6.864261e+01  5.678472e+01   
min         1.400000e+01   -2.500000e+01  0.000000e+00 -1.090000e+02   
25%         8.000000e+01    8.200000e+01  5.800000e+01  9.000000e+00   
50%         1.160000e+02    1.160000e+02  9.000000e+01  2.400000e+01   
75%         1.650000e+02    1.650000e+02  1.370000e+02  5.600000e+01   
max         1.114000e+03    6.600000e+02  1.091000e+03  2.461000e+03   

           DepDelay      Distance        TaxiIn       TaxiOut     Cancelled  \
count  1.936758e+06  1.936758e+06  1.929648e+06  1.936303e+06  1.936758e+06   
mean   4.318518e+01  7.656862e+02  6.812975e+00  1.823220e+01  3.268348e-04   
std    5.340250e+01  5.744797e+02  5.273595e+00  1.433853e+01  1.807562e-02   
min    6.000000e+00  1.100000e+01  0.000000e+00  0.000000e+00  0.000000e+00   
25%    1.200000e+01  3.380000e+02  4.000000e+00  1.000000e+01  0.000000e+00   
50%    2.400000e+01  6.060000e+02  6.000000e+00  1.400000e+01  0.000000e+00   
75%    5.300000e+01  9.980000e+02  8.000000e+00  2.100000e+01  0.000000e+00   
max    2.467000e+03  4.962000e+03  2.400000e+02  4.220000e+02  1.000000e+00   

           Diverted  CarrierDelay  WeatherDelay      NASDelay  SecurityDelay  \
count  1.936758e+06  1.247488e+06  1.247488e+06  1.247488e+06   1.247488e+06   
mean   4.003598e-03  1.917940e+01  3.703571e+00  1.502164e+01   9.013714e-02   
std    6.314722e-02  4.354621e+01  2.149290e+01  3.383305e+01   2.022714e+00   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   0.000000e+00   
25%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   0.000000e+00   
50%    0.000000e+00  2.000000e+00  0.000000e+00  2.000000e+00   0.000000e+00   
75%    0.000000e+00  2.100000e+01  0.000000e+00  1.500000e+01   0.000000e+00   
max    1.000000e+00  2.436000e+03  1.352000e+03  1.357000e+03   3.920000e+02   

       LateAircraftDelay  
count       1.247488e+06  
mean        2.529647e+01  
std         4.205486e+01  
min         0.000000e+00  
25%         0.000000e+00  
50%         8.000000e+00  
75%         3.300000e+01  
max         1.316000e+03

In [67]:
# S'utilitza la primera columna com a variable identificadora
data_air = data_airlines.rename(columns={"Unnamed: 0": "ID"})
data_air=data_air.set_index('ID')
data_air

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
ID                                                                          
0        2008      1           3          4   2003.0        1955   2211.0   
1        2008      1           3          4    754.0         735   1002.0   
2        2008      1           3          4    628.0         620    804.0   
4        2008      1           3          4   1829.0        1755   1959.0   
5        2008      1           3          4   1940.0        1915   2121.0   
...       ...    ...         ...        ...      ...         ...      ...   
7009710  2008     12          13          6   1250.0        1220   1617.0   
7009717  2008     12          13          6    657.0         600    904.0   
7009718  2008     12          13          6   1007.0         847   1149.0   
7009726  2008     12          13          6   1251.0        1240   1446.0   
7009727  2008     12          13          6   1110.0        1103   1413.0   

         CRSArrTime UniqueCarrier  FlightNum TailNum  ActualElapsedTime  \
ID                                                                        
0              2225            WN        335  N712SW              128.0   
1              1000            WN       3231  N772SW              128.0   
2               750            WN        448  N428WN               96.0   
4              1925            WN       3920  N464WN               90.0   
5              2110            WN        378  N726SW              101.0   
...             ...           ...        ...     ...                ...   
7009710        1552            DL       1621  N938DL              147.0   
7009717         749            DL       1631  N3743H              127.0   
7009718        1010            DL       1631  N909DA              162.0   
7009726        1437            DL       1639  N646DL              115.0   
7009727        1418            DL       1641  N908DL              123.0   

         CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  \
ID                                                                           
0                 150.0    116.0     -14.0       8.0    IAD  TPA       810   
1                 145.0    113.0       2.0      19.0    IAD  TPA       810   
2                  90.0     76.0      14.0       8.0    IND  BWI       515   
4                  90.0     77.0      34.0      34.0    IND  BWI       515   
5                 115.0     87.0      11.0      25.0    IND  JAX       688   
...                 ...      ...       ...       ...    ...  ...       ...   
7009710           152.0    120.0      25.0      30.0    MSP  ATL       906   
7009717           109.0     78.0      75.0      57.0    RIC  ATL       481   
7009718           143.0    122.0      99.0      80.0    ATL  IAH       689   
7009726           117.0     89.0       9.0      11.0    IAD  ATL       533   
7009727           135.0    104.0      -5.0       7.0    SAT  ATL       874   

         TaxiIn  TaxiOut  Cancelled CancellationCode  Diverted  CarrierDelay  \
ID                                                                             
0           4.0      8.0          0                N         0           NaN   
1           5.0     10.0          0                N         0           NaN   
2           3.0     17.0          0                N         0           NaN   
4           3.0     10.0          0                N         0           2.0   
5           4.0     10.0          0                N         0           NaN   
...         ...      ...        ...              ...       ...           ...   
7009710     9.0     18.0          0                N         0           3.0   
7009717    15.0     34.0          0                N         0           0.0   
7009718     8.0     32.0          0                N         0           1.0   
7009726    13.0     13.0          0                N         0           NaN   
7009727     8.0     11.0          0                N    

In [68]:
# S'eliminen les variables que no són d'interès
data_air = data_air.drop(['TailNum','ActualElapsedTime', 'CRSElapsedTime','TaxiIn','TaxiOut','Diverted','CarrierDelay', 'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay','Cancelled','CancellationCode'], axis=1)
data_air[:3]

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
ID                                                                     
0   2008      1           3          4   2003.0        1955   2211.0   
1   2008      1           3          4    754.0         735   1002.0   
2   2008      1           3          4    628.0         620    804.0   

    CRSArrTime UniqueCarrier  FlightNum  AirTime  ArrDelay  DepDelay Origin  \
ID                                                                            
0         2225            WN        335    116.0     -14.0       8.0    IAD   
1         1000            WN       3231    113.0       2.0      19.0    IAD   
2          750            WN        448     76.0      14.0       8.0    IND   

   Dest  Distance  
ID                 
0   TPA       810  
1   TPA       810  
2   BWI       515

In [69]:
# Es converteix en string les variables Month, DayofMonth, DayofWeek i FlightNum per treballar amb el groupby més endevant
data_air['Year'] = data_air['Year'].apply(str)
data_air['Month'] = data_air['Month'].apply(str)
data_air['DayofMonth'] = data_air['DayofMonth'].apply(str)
data_air['DayOfWeek'] = data_air['DayOfWeek'].apply(str)
data_air['FlightNum'] = data_air['FlightNum'].apply(str)
data_air[:3]

Year Month DayofMonth DayOfWeek  DepTime  CRSDepTime  ArrTime  CRSArrTime  \
ID                                                                              
0   2008     1          3         4   2003.0        1955   2211.0        2225   
1   2008     1          3         4    754.0         735   1002.0        1000   
2   2008     1          3         4    628.0         620    804.0         750   

   UniqueCarrier FlightNum  AirTime  ArrDelay  DepDelay Origin Dest  Distance  
ID                                                                             
0             WN       335    116.0     -14.0       8.0    IAD  TPA       810  
1             WN      3231    113.0       2.0      19.0    IAD  TPA       810  
2             WN       448     76.0      14.0       8.0    IND  BWI       515

## Exercici 2
__Fes un informe complet del data set:__

- Resumeix estadísticament les columnes d’interès

In [70]:
# Dades numèriques. S'ignora les columnes DepTime, CRSDepTime, ArrTime i CRSArrTime ja que es tracta de variables que 
# representen l'hora
data_air[['AirTime','ArrDelay','DepDelay','Distance']].describe().round(3)

AirTime     ArrDelay     DepDelay     Distance
count  1928371.000  1928371.000  1936758.000  1936758.000
mean       108.277       42.200       43.185      765.686
std         68.643       56.785       53.403      574.480
min          0.000     -109.000        6.000       11.000
25%         58.000        9.000       12.000      338.000
50%         90.000       24.000       24.000      606.000
75%        137.000       56.000       53.000      998.000
max       1091.000     2461.000     2467.000     4962.000

<span style="color:blue">
    <p>AirTime: la durada mitjana dels vols es de 108 minut. Hi ha vols de 0 minuts (els vols cancel·lats). 
       El vol que més ha durat és de 1091 minuts (unes 18 hores)</p>
    <p>ArrDelay: L'endarreriment en l'aterratge promig és de durada mitjana dels vols es de 42 minuts. 
       Hi ha un vol que ha arribat 109 minuts abans de l'hora prevista (pot ser es tracta d'una anomalia o d'un outlier).  
       El major endarreriment ha estat de 2461 minuts, és a dir,  més d'un dia i mig</p>
    <p>Distance: la distancia mitja dels vols es de 765 milles. El trajecte més curt es d'11 milles i el més llarg de 4962.</p>
</span>    

In [71]:
# Dades categòriques. Es compten les observacions, el valors únics, el valor més repetit (top) i la freqüencia del
# valor més repetit
data_air.describe(include=np.object)

Year    Month DayofMonth DayOfWeek UniqueCarrier FlightNum  \
count   1936758  1936758    1936758   1936758       1936758   1936758   
unique        1       12         31         7            20      7499   
top        2008       12         22         5            WN        16   
freq    1936758   203385      71649    323259        377602      1586   

         Origin     Dest  
count   1936758  1936758  
unique      303      304  
top         ATL      ORD  
freq     131613   108984

<span style="color:blue">
    <p>UniqueCarrier: apareixen 20 aerolinies, i la més utilitzada es WN</p>
    <p>Origin: apareixen 303 llocs d'origen diferents. ATL és el lloc amb més sortides de vols</p>
    <p>Dest: apareixen 304 llocs de destí diferents. ORD és el lloc amb més arribades de vols.</p>
</span>   

- Troba quantes dades faltants hi ha per columna

In [72]:
# Es compten els missings que hi ha en cada variable:
data_air.isnull().sum()

Year                0
Month               0
DayofMonth          0
DayOfWeek           0
DepTime             0
CRSDepTime          0
ArrTime          7110
CRSArrTime          0
UniqueCarrier       0
FlightNum           0
AirTime          8387
ArrDelay         8387
DepDelay            0
Origin              0
Dest                0
Distance            0
dtype: int64

<span style='color:blue'>Tot i que caldria tractar els missings (imputació o eliminació justificada), es decideix no
    modificar la base de dades ja que aquest tema no pertoca encara i tampoc afecta als mètodes següents que s'apliquen.
</span>

- Crea columnes noves (velocitat mitjana del vol, si ha arribat tard o no...)

In [73]:
# Distancia en KM
data_air['Dist_km'] = round(data_air['Distance']*1.60934,2)

In [74]:
# Velocitat mitjana (Km/h)
data_air['VelMitjana'] = round(data_air['Dist_km']/(data_air['AirTime']/60),2)

In [75]:
# L'avió ha arribat tard?
# Es considera que un avió arriba tard si aterra 5 minuts més tard de l'hora prevista.
# Es treballa amb la variable ArrDelay
data_air['Tard_Arr'] = np.where(data_air['ArrDelay']<=5, 'No', 'Yes')

In [76]:
# S'utilitza la variable CRSDepTime per crear una nova variable que indica si el vol programat surt de matí (de 4:01 a 12:00), 
# de tarda (de 12:00 a 20:00) o de nit (de 20:01 a 4:00)
data_air['Sortida'] = np.where((data_air['CRSDepTime'] > 1200) & (data_air['CRSDepTime'] <= 2000), 'Tarda',
                               np.where((data_air['CRSDepTime'] > 400) & (data_air['CRSDepTime'] <= 1200), 'Mati', 'Nit'))

In [77]:
# Visualització del dataframe amb les noves variables
data_air

Year Month DayofMonth DayOfWeek  DepTime  CRSDepTime  ArrTime  \
ID                                                                       
0        2008     1          3         4   2003.0        1955   2211.0   
1        2008     1          3         4    754.0         735   1002.0   
2        2008     1          3         4    628.0         620    804.0   
4        2008     1          3         4   1829.0        1755   1959.0   
5        2008     1          3         4   1940.0        1915   2121.0   
...       ...   ...        ...       ...      ...         ...      ...   
7009710  2008    12         13         6   1250.0        1220   1617.0   
7009717  2008    12         13         6    657.0         600    904.0   
7009718  2008    12         13         6   1007.0         847   1149.0   
7009726  2008    12         13         6   1251.0        1240   1446.0   
7009727  2008    12         13         6   1110.0        1103   1413.0   

         CRSArrTime UniqueCarrier FlightNum  AirTime  ArrDelay  DepDelay  \
ID                                                                         
0              2225            WN       335    116.0     -14.0       8.0   
1              1000            WN      3231    113.0       2.0      19.0   
2               750            WN       448     76.0      14.0       8.0   
4              1925            WN      3920     77.0      34.0      34.0   
5              2110            WN       378     87.0      11.0      25.0   
...             ...           ...       ...      ...       ...       ...   
7009710        1552            DL      1621    120.0      25.0      30.0   
7009717         749            DL      1631     78.0      75.0      57.0   
7009718        1010            DL      1631    122.0      99.0      80.0   
7009726        1437            DL      1639     89.0       9.0      11.0   
7009727        1418            DL      1641    104.0      -5.0       7.0   

        Origin Dest  Distance  Dist_km  VelMitjana Tard_Arr Sortida  
ID                                                                   
0          IAD  TPA       810  1303.57      674.26       No   Tarda  
1          IAD  TPA       810  1303.57      692.16       No    Mati  
2          IND  BWI       515   828.81      654.32      Yes    Mati  
4          IND  BWI       515   828.81      645.83      Yes   Tarda  
5          IND  JAX       688  1107.23      763.61      Yes   Tarda  
...        ...  ...       ...      ...         ...      ...     ...  
7009710    MSP  ATL       906  1458.06      729.03      Yes   Tarda  
7009717    RIC  ATL       481   774.09      595.45      Yes    Mati  
7009718    ATL  IAH       689  1108.84      545.33      Yes    Mati  
7009726    IAD  ATL       533   857.78      578.28      Yes   Tarda  
7009727    SAT  ATL       874  1406.56      811.48       No    Mati  

[1936758 rows x 20 columns]

- Taula de les aerolínies amb més endarreriments acumulats

In [78]:
# Es crea una nova base de dades filtrant les observacions que tenen endarreriment a l'arribada (Tard_Arr = "Yes")
data_air_tard = data_air[data_air['Tard_Arr']=='Yes'] 

In [79]:
# Endarreriments acumulats per aerolínia:
data_end = data_air_tard[['UniqueCarrier','Tard_Arr']].groupby(['UniqueCarrier']).count()
data_end

Tard_Arr
UniqueCarrier          
9E                43426
AA               160134
AQ                  572
AS                31133
B6                45161
CO                75040
DL                92334
EV                69536
F9                22756
FL                59674
HA                 6571
MQ               120717
NW                67427
OH                46504
OO               112021
UA               114674
US                75465
WN               282343
XE                87534
YV                60188

In [80]:
# Les 5 aerolínies amb més endarreriments:
data_end.sort_values(by='Tard_Arr',ascending=False)[:5]

Tard_Arr
UniqueCarrier          
WN               282343
AA               160134
MQ               120717
UA               114674
OO               112021

- Quins són els vols més llargs? I els més endarrerits?

In [81]:
# Com que hi ha números de vol que comparteixen el mateix recorregut, els vols més llargs serien tots aquells que 
# tinguessin la distancia màxima. Per això es decideix trobar els 5 recorreguts més llargs i assignar-los un
# dels múltiples vols que hi ha.

# Es crea una taula amb el número de vol, l'origen, el destí i la distància entre ambdós llocs
# Com que hi ha vols que tenen el mateix recorregut però invertint els origens i destins, s'agafa un dels dos possibles
# i se li assigna un número de vol corresponent
data_v_ll = data_air[['FlightNum','Origin','Dest','Distance']].groupby(['Distance']).first().reset_index()
data_v_ll.sort_values(by=['Distance'],ascending=False)[:5]

Distance FlightNum Origin Dest
1418      4962        15    EWR  HNL
1417      4502       850    HNL  ATL
1416      4243         1    ORD  HNL
1415      4213         3    KOA  ORD
1414      4184         3    ORD  OGG

In [82]:
# Els 5 vols més endarrerits (respecte l'aterratge, en minuts):
data_air[['FlightNum','ArrDelay']].sort_values(by='ArrDelay',ascending=False)[:5]

FlightNum  ArrDelay
ID                         
1018798       808    2461.0
2235378      1699    2453.0
2832617      1107    1951.0
3387883      3538    1707.0
6857047       357    1655.0

- Altres questions:

In [83]:
# Moment del dia amb més sortides d'avions
data_md = data_air[['Year','Sortida']].groupby(['Sortida']).count().sort_values(by = ['Year'], ascending=False)
data_md = data_md.rename(columns={'Year' : 'Count'}) 
data_md

Count
Sortida         
Tarda    1174187
Mati      537744
Nit       224827

In [84]:
# Els 5 dies amb més sortides de vols durant l'any 2008
data_5d = data_air[['Year','Month','DayofMonth']].groupby(['Month','DayofMonth']).count()
data_5d.sort_values(by = ['Year'], ascending=False)[:5]

Year
Month DayofMonth       
1     2           10857
12    22          10693
      23          10621
      1           10367
      21           9994

 <span style="color:blue">Els dies amb més vols coincideixen amb les vacances d'hivern.</span>

In [85]:
# Els dos mesos amb més endarreriments per vol (freqüència relativa)

## 1) Es calcula la freqüència absoluta de vols per mes
freq_vol_mes = data_air[['Year','Month']].groupby(['Month']).count()
freq_vol_mes = freq_vol_mes.rename(columns={'Year' : 'Freq_abs'})
freq_vol_mes

Freq_abs
Month          
1        183527
10       103525
11       105563
12       203385
2        189534
3        200842
4        155264
5        153072
6        200914
7        182945
8        162648
9         95539

In [86]:
## 2) Es calcula la freqüència absoluta de vols endarrerits (més de 5 minuts a l'aterrar) per mes
freq_vol_end = data_air_tard[['Year','Month']].groupby(['Month']).count()
freq_vol_end = freq_vol_end.rename(columns={'Year' : 'Freq_abs_end'})
freq_vol_end

Freq_abs_end
Month              
1            150489
10            77488
11            81872
12           171478
2            158461
3            164600
4            125347
5            122112
6            168792
7            149924
8            128978
9             73669

In [87]:
## 3) Es calcula la freqüència relativa d'endarreriments per mes
data_mend = pd.concat([freq_vol_end, freq_vol_mes], axis=1)
data_mend['Freq_rel_end']=data_mend['Freq_abs_end']/data_mend['Freq_abs']
data_mend

Freq_abs_end  Freq_abs  Freq_rel_end
Month                                      
1            150489    183527      0.819983
10            77488    103525      0.748496
11            81872    105563      0.775575
12           171478    203385      0.843120
2            158461    189534      0.836056
3            164600    200842      0.819550
4            125347    155264      0.807315
5            122112    153072      0.797742
6            168792    200914      0.840121
7            149924    182945      0.819503
8            128978    162648      0.792989
9             73669     95539      0.771088

In [88]:
## 4) S'ordena per la freqüencia relativa d'endarreriment de major a menor i s'agafen les dues primeres observacions 
data_mend.sort_values(by = ['Freq_rel_end'], ascending=False)[:2]

Freq_abs_end  Freq_abs  Freq_rel_end
Month                                      
12           171478    203385      0.843120
6            168792    200914      0.840121

<span style='color:blue'>Els mesos amb més endarreriments relatius són juny i dessembre, que són els mesos de començament de vacances.</span>

In [89]:
# Els 3 primers origens amb més endarreriments
data_o_end = data_air_tard[['Year','Origin']].groupby(['Origin']).count()
data_o_end = data_o_end.rename(columns={'Year' : 'Endarreriments'}) 
data_o_end.sort_values(by='Endarreriments',ascending=False)[0:3]

Endarreriments
Origin                
ATL             110798
ORD             107214
DFW              79283

## Exercici 3
__Exporta el data set net i amb les noves columnes a Excel.__

<span style='color:blue'>
    <br>Com que l'arxiu és massa gran, es decideix exportar-lo a csv.</br>
</span>

In [98]:
data_air.to_csv('Vols_endarrerits.csv')